fuel-tx/src/transaction.rs

```diff
@@ -733,6 +734,25 @@ pub mod field {
 
         /// Returns predicate's offset and length of the `Input` at `idx`, if any.
         fn inputs_predicate_offset_at(&self, idx: usize) -> Option<(usize, usize)>;
+
+        /// Returns all signed inputs. This excludes predicate inputs.
+        fn signed_inputs(&self) -> Vec<&Input> {
+            self.inputs()
+                .iter()
+                .filter_map(|input| input.witness_index().is_some().then_some(input))
+                .collect_vec()
+        }
+
+        /// Returns all signed inputs filtered by witness uniqueness.
+        fn signed_inputs_with_unique_witnesses(&self) -> Vec<&Input> {
+            self.signed_inputs()
+                .into_iter()
+                .sorted_unstable_by(|a, b| {
+                    Ord::cmp(&a.witness_index(), &b.witness_index())
+                })
+                .dedup()
+                .collect_vec()
+        }
```

Comment:
It is better to not add one-time functions into the trait. We only need them in one place, plus we need to iterate over all inputs because you need to charge for `Signed` and for `Predicate` inputs

Reply:
I understand your suggestion. Instead specializing the trait here, I can just iterate over the inputs at the time of calculating the fee, as you suggest in another comment. 

Currently, this method is also used in tests to verify the fee for a transaction is correct. But maybe I can refactor the tests so that we can test the fees without duplicating the calculation logic, and we can keep the calculation isolated to the one place.